packages/edit-site/src/components/global-styles/screen-style-variations.js

```diff
@@ -120,6 +129,31 @@ function ScreenStyleVariations() {
 		];
 	}, [ variations ] );
 
+	const { __unstableSetEditorMode } = useDispatch( blockEditorStore );
+	const shouldRevertInitialMode = useRef( null );
+	useEffect( () => {
```

Comment:
We're tracking mode changes to decide whether to revert to the initial mode or not. If the user has done any mode changes since the style variations were opened, we won't revert to the initial.

So, in a situation like this:

- user starts in edit mode
- goes to style variations, so the editor is switched automatically to zoom-out mode
- user intentionally switches off zoom-out mode (goes back to edit)
- user intentionally switches on zoom-out mode
- user closes global styles sidebar

What's the expected mode at this point?

With this PR, the mode stays in zoom-out. I think it makes sense because trying to be _too_ smart may be too intrusive. Though, I see how switching off zoom-out mode can be something people do if they want _more_ focus for a specific change, and then going back to the zoom-out.

Reply:
I opted for this approach because it was the most simple code and seemed to fit most use cases. If you prefer, I can try an approach where if the end mode is still zoom-out we always revert to the initial mode.